api/envoy/config/listener/v3/listener.proto

```diff
@@ -233,7 +233,10 @@ message Listener {
   google.protobuf.BoolValue freebind = 11;
 
   // Additional socket options that may not be present in Envoy source code or
-  // precompiled binaries.
+  // precompiled binaries. The socket options can be updated for a listener and
+  // it only supports when the
+  // :ref:`enable_reuse_port <envoy_v3_api_field_config.listener.v3.Listener.enable_reuse_port>`
+  // is `true`.
```

Comment:
```suggestion
  // precompiled binaries. The socket options can be updated for a listener
  // only when
  // :ref:`enable_reuse_port <envoy_v3_api_field_config.listener.v3.Listener.enable_reuse_port>`
  // is `true`. Otherwise, if socket options change during a listener update the updated will be NACKd 
  // to make it clear that the options were not updated.
```

We should also put the new behavior behind a runtime flag as this might cause people issues if they didn't know they were ignoring changes to configs.